src/app/_components/BlogSortController.tsx

```diff
@@ -0,0 +1,26 @@
+'use client'
+
+import { useState, useEffect } from 'react'
+
+import { useUpdateSearchParams } from '@/hooks/useUpdateSearchParams'
+
+import { SortListbox } from '@/components/SortListbox'
+
+import { type SortOptionType } from '@/types/sortTypes'
+
+import { SORT_KEY } from '@/constants/searchParams'
+import { DEFAULT_SORT_OPTION } from '@/constants/sortConstants'
+
+export function BlogSortController() {
+  const [sortOption, setSortOption] =
+    useState<SortOptionType>(DEFAULT_SORT_OPTION)
+  const updateSearchParams = useUpdateSearchParams()
+
+  useEffect(() => {
+    updateSearchParams({ [SORT_KEY]: sortOption })
+  }, [sortOption, updateSearchParams])
```

Comment:
This sets the `sort` search params on page load. `Pagination` only updates the search params on the first user interaction. 

I'm not sure which one is best, but we should agree on one behaviour and normalise it for all components updating the URL.

We don't have to implement the changes now; just make sure to include them in a ticket.

PS: Search is different as there's no default value.